specs/derivation.md

```diff
@@ -587,7 +612,9 @@ In particular, the following fields of the payload attributes are checked for eq
 If consolidation fails, the unsafe L2 head is reset to the safe L2 head.
 
 If the safe and unsafe L2 heads are identical (whether because of failed consolidation or not), we send the block to the
-execution engine to be converted into a proper L2 block, which becomes both the new L2 safe and unsafe heads.
+execution engine to be converted into a proper L2 block, which will become the new L2 unsafe head.
+
+TODO TODO TODO: above said "will become both new L2 safe and unsafe heads" -- does the code do this?
```

Comment:
yes. when sending a payload building on top of the unsafe/safe head (if they are the same), we advance both. We never let the unsafe head get behind the safe head.